streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBRangeIterator.java

```diff
@@ -32,34 +34,49 @@
     private final byte[] rawLastKey;
     private final boolean forward;
     private final boolean toInclusive;
+    private final ByteBuffer directByteBuffer;
 
     RocksDBRangeIterator(final String storeName,
                          final RocksIterator iter,
                          final Set<KeyValueIterator<Bytes, byte[]>> openIterators,
                          final Bytes from,
                          final Bytes to,
                          final boolean forward,
-                         final boolean toInclusive) {
+                         final boolean toInclusive,
+                         final ByteBuffer directByteBuffer) {
         super(storeName, iter, openIterators, forward);
         this.forward = forward;
         this.toInclusive = toInclusive;
+        this.directByteBuffer = directByteBuffer;
         if (forward) {
             if (from == null) {
                 iter.seekToFirst();
             } else {
-                iter.seek(from.get());
+                allocateDirectByteBufferAndSeek(iter, from, true);
             }
             rawLastKey = to == null ? null : to.get();
         } else {
             if (to == null) {
                 iter.seekToLast();
             } else {
-                iter.seekForPrev(to.get());
+                allocateDirectByteBufferAndSeek(iter, to, false);
             }
             rawLastKey = from == null ? null : from.get();
         }
     }
 
+    private void allocateDirectByteBufferAndSeek(final RocksIterator iter, final Bytes bytes, final boolean forward) {
+        /*final ByteBuffer directByteBuffer = ByteBuffer.allocateDirect(bytes.get().length);
+        directByteBuffer.order(ByteOrder.nativeOrder());*/
+        directByteBuffer.clear();
+        directByteBuffer.put(bytes.get());
+        directByteBuffer.flip();
```

Comment:
You might want to pull this up or into a separate method to avoid checking the direction twice